python/hail/matrixtable.py

```diff
@@ -842,18 +841,24 @@ def select_globals(self, *exprs, **named_exprs):
         strs = []
         all_exprs = []
         base, cleanup = self._process_joins(*itertools.chain(exprs, named_exprs.values()))
+
+        ids = []
         for e in exprs:
             all_exprs.append(e)
             analyze('MatrixTable.select_globals', e, self._global_indices)
             if e._ast.search(lambda ast: not isinstance(ast, Reference) and not isinstance(ast, Select)):
                 raise ExpressionException("method 'select_globals' expects keyword arguments for complex expressions")
             strs.append(
-                '{}: {}'.format(e._ast.selection if isinstance(e._ast, Select) else e._ast.name, e._ast.to_hql()))
+                '{}: {}'.format(e._ast.name if isinstance(e._ast, Select) else e._ast.name, e._ast.to_hql()))
+            ids.append(e._ast.expand()[0].name)
```

Comment:
Don't you just want `e._ast.name` here?  Namely, the name you're putting into strs in the line above?

Also, don't you need to escape the name in the line above?

Reply:
yeah, that sounds right. I didn't even see this code.